image.yaml

```diff
@@ -18,6 +24,13 @@ labels:
     - name: "io.openshift.s2i.scripts-url"
       value: "image:///usr/local/s2i"
 envs:
+    - name: "JBOSS_PRODUCT"
+      value: *product
+    - name: "JBOSS_SSO_VERSION"
+      value: *product_version
+    - name: "PRODUCT_VERSION"
+      value: *product_version
+
```

Comment:
Similarly for these -- should the:

- `JAVA_HOME`,
- `JAVA_VENDOR`,
- `JAVA_VERSION`,
- `JBOSS_CONTAINER_OPENJDK_JDK_MODULE`,
- `JBOSS_EAP_VERSION`,
- `JBOSS_IMAGE_NAME`,
- `JBOSS_IMAGE_VERSION`, 
- `JBOSS_CONTAINER_JAVA_PROXY_MODULE`, 
-`JBOSS_CONTAINER_JAVA_JVM_MODULE`,
- `JBOSS_CONTAINER_MAVEN_35_MODULE`, 
- `MAVEN_VERSION`,
- `JBOSS_CONTAINER_UTIL_LOGGING_MODULE`,
- `JBOSS_CONTAINER_MAVEN_DEFAULT_MODULE`,
- `S2I_SOURCE_DEPLOYMENTS_FILTER`,
- `JBOSS_CONTAINER_S2I_CORE_MODULE`,
- `JBOSS_CONTAINER_EAP_S2I_MODULE`,
- `DEFAULT_ADMIN_USERNAME`,
- `JBOSS_MODULES_SYSTEM_PKGS`

be defined too?

_Disclaimer_: At the moment unable to build the image for testing locally neither using docker nor podman engines, so unable to verify if EAP / JDK layers would bring those variables as an intermediary result already. But at least the `DEFAULT_ADMIN_USERNAME` & `DEFAULT_ADMIN_USERNAME` seem to be used by the EAP layer for configuration, unsure about the others.

From consistency PoV it seems to me those env vars should be preserved in new RH-SSO release stream too AFAICT.

Reply:
Per discussion with @hmlnarik  env vars listed above should be defined / provided by modules listed in the image.